.github/workflows/ci.yml

```
@@ -42,6 +40,23 @@ jobs:
         with:
           token: ${{ secrets.GH_TOKEN || secrets.GITHUB_TOKEN }}
 
+      - name: Semantic Release
+        uses: cycjimmy/semantic-release-action@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+          SEMANTIC_RELEASE_PACKAGE: ${{ github.event.repository.name }}
+        with:
+          extra_plugins: |
+            @semantic-release/changelog
+            @semantic-release/git
+
+      - name: Checkout tag
+        if: ${{ steps.semantic.outputs.new_release_version }} > ${{ steps.semantic.outputs.last_release_version }}
```

Not sure this kind of comparison will work, since it's comparing strings and not numbers (`1.0.1 > 1.0.0 vs 101 > 100)`. I would suggest changing this:
```suggestion
        if: steps.semantic.outputs.new_release_version != ''
```

If there is no new release created, then `steps.semantic.outputs.new_release_version` will be empty and this step will be skipped. Also, the `${{ }}` brackets are optional for the `if` property in an action.

---

changes made
also had to add
```id: semantic```
to enable outputs from the semantic release task